assets/js/modules/analytics-4/components/common/PropertySelect.js

```diff
@@ -88,6 +88,34 @@ export default function PropertySelect( {
 	);
 
 	const { selectProperty } = useDispatch( MODULES_ANALYTICS_4 );
+
+	const measurementIDs = useSelect( ( select ) => {
+		// We should conditionally check the properties length here
+		// because the properties will be null if the user does not have access to the module.
+		if ( ! properties?.length ) {
+			return null;
+		}
+		return properties.reduce( ( acc, property ) => {
+			const currentPropertyID = property._id;
+			const dataStream =
+				select( MODULES_ANALYTICS_4 ).getWebDataStreamsBatch(
+					currentPropertyID
+				);
+			if ( ! Object.keys( dataStream ).length ) {
+				isLoading = true;
```

Comment:
There is a problem here. If we delete `googlesitekit_analytics-4_settings` from the wp_options table, `getWebDataStreamsBatch` selector will return an empty object. Hence the following condition will be `true`. So, the loader is displayed all the time. Please refer to the recording below:

https://user-images.githubusercontent.com/24408261/195570652-e2c095ae-cc50-4a5e-88aa-d0f5466ec3bb.mov

We shouldn't set the loader to `true` here to avoid the above issue. Rather we should show the property select dropdown without the property being selected:

![Screenshot 2022-10-13 at 3 37 19 PM](https://user-images.githubusercontent.com/24408261/195571180-f25fb057-750c-422f-bcc1-c453275b9899.png)


To make it work, we should conditionally get the measurement ID from the object `measurementIDs?.[ _id ]` in the `Option` element.

WDYT, @aaemnnosttv @tofumatt?

Reply:
Getting the measurement ID from `measurementIDs?.[ _id ]` makes sense to me 👍🏻

Reply:
Thanks, @tofumatt